core/orchestrator.go

```diff
@@ -628,6 +628,7 @@ func (n *LivepeerNode) transcodeSegmentLoop(md *SegTranscodingMetadata, segChan
 				// timeout; clean up goroutine here
 				os.EndSession()
 				los.EndSession()
+				delete(n.TranscoderManager.streamSessions, md.AuthToken.SessionId)
```

Comment:
There can be multiple segments (video chunks) in flight that access this map for a particular key, ideally we thus want this to be concurrency safe. 

Let's wrap this `delete` statement in a method that also locks and unlocks the resource. 

```go
func (rtm *RemoteTranscoderManager) deleteSession(sesssionId string) {
  rtm.RTmutex.Lock()
  defer rtm.RTmutex.Unlock()
  delete(n.TranscoderManager.streamSessions, md.AuthToken.SessionId)
}
```

We can then do `n.TranscoderManager.deleteSession(md.AuthToken.SessionId)`

Reply:
On second thought let's add the delete statement to https://github.com/livepeer/go-livepeer/blob/7a0fd1f6b958d61177fc01c488972f54b8bad1e3/core/orchestrator.go#L878

Reply:
Thanks for reviewing, @kyriediculous. Will take care of this, I was almost certain that the cleanup would have to be handled in other places as well.